### PR TITLE
refactor(repositories): item flag extends immutable

### PR DESCRIPTION
**Pull request by Defferrard**

### CODE REVIEW

**Review by ReidyT — APPROVED** [review]
LGTM 🧑‍🚀

File: src/services/item/plugins/itemFlag/index.ts [hunk 1]
```diff
@@ -20,25 +21,21 @@ const plugin: FastifyPluginAsync = async (fastify) => {
...
 
   // get flags
-  fastify.get(
-    '/flags',
-    { schema: getFlags, preHandler: optionalIsAuthenticated },
-    async ({ user }) => {
-      return itemFlagService.getAllFlags(user?.account, buildRepositories());
-    },
-  );
+  fastify.get('/flags', { schema: getFlags }, async () => {
```
Comment (spaenleh) [inline src/services/item/plugins/itemFlag/index.ts:24]: I see you removed the requirement for authentication, what is the plan for this endpoint ?

**Review by spaenleh — APPROVED** [review]

File: src/services/item/plugins/itemFlag/index.ts (hunk 1 above)
Comment (spaenleh) [reply src/services/item/plugins/itemFlag/index.ts:24]: Mhmmm... okay I thought this was returning all the elements that were flagged, But in reality it just returns the static array of flag types. Maybe adding a small comment would help disambiguate ? In fine we should probably change the name of the endpoint, maybe `/flagTypes` ?

Comment (Defferrard) [reply src/services/item/plugins/itemFlag/index.ts:24]: I agree ! I'd name that route `GET /flags/types` I've added this route to https://github.com/graasp/graasp/issues/1288

**Pull request merged**